src/magic_spell_effect.cpp

```diff
@@ -481,6 +482,69 @@ void spell_effect::target_attack( const spell &sp, Creature &caster,
     }
 }
 
+static void magical_polymorph( monster &victim, Creature &caster, const spell &sp )
+{
+    std::string new_id = sp.effect_data();
```

Comment:
Would be easier to read and to follow if you change this to `mtype_id`, so you would not need to cast this string to a type id all the time in the code below.